File: rest-service/src/main/java/mb/lib/query/model/FailedQueryJobResponse.java
package mb.lib.query.model;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonSetter;
import mb.lib.queue.consts.JsonKeys;
import mb.lib.queue.model.FailedJobResponse;

public class FailedQueryJobResponse implements FailedJobResponse<FailedQueryJob>
{
  private List<FailedQueryJob> failedJobs;

  @JsonGetter(JsonKeys.FailedJobs)
  public List<FailedQueryJob> getFailedJobs() {
    return failedJobs;
  }

  @JsonSetter(JsonKeys.FailedJobs)
  public FailedJobResponse<FailedQueryJob> setFailedJobs(List<FailedQueryJob> failedJobs) {
    this.failedJobs = failedJobs;
    return this;
  }
}

File: rest-service/src/main/kotlin/mb/lib/query/model/FailedQueryJobResponse.kt
package mb.lib.query.model

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.JsonNode
import mb.lib.queue.consts.JsonKeys
import mb.lib.queue.model.FailedJobResponse
import mb.lib.util.jsonCast
import mb.lib.util.jsonObject

data class FailedQueryJobResponse(
  @JsonProperty(JsonKeys.FailedJobs) override val failedJobs: List<FailedQueryJob>
) : FailedJobResponse<FailedQueryJob> {
  override fun toJSON() = jsonObject { set<JsonNode>(JsonKeys.FailedJobs, failedJobs.jsonCast()) }
}
